lib/Doctrine/DBAL/Platforms/DB2Platform.php

```diff
@@ -878,4 +880,64 @@ protected function getReservedKeywordsClass()
     {
         return Keywords\DB2Keywords::class;
     }
+    
+    /**
+     * Prepare ORDER BY string for OVER() if applicable
+     * 
+     * @param  string $query
+     * 
+     * @return array|string[]|mixed[]
+     */
+    private function getOrderByForOver($query)
+    {
+        //preset empty array
+        $orderByArray = [];
+
+        // remove extra white space
+        $query = preg_replace('/\s+/', ' ', $query);
+
+        //determine if 'ORDER BY' is part of the query
+        $orderByFound = preg_match('/(\bORDER\b)(\s*)(\bBY\b)(?!.*\b\1\b)/i', $query, $orderByPositionArray, PREG_OFFSET_CAPTURE);
+
+        // ORDER BY found in query string
+        if (0 !== $orderByFound) {
+            $orderByPosition = $orderByPositionArray[0][1];
+
+            $queryArray = preg_split('/[, ]/', substr($query, 0, $orderByPosition -1));
+            $orderByArray = explode(',', substr($query, $orderByPosition));
+
+            foreach ($orderByArray as $orderIndex => $orderValue) {
+                $splitOrder = explode(' ', $orderValue);
+
+                foreach ($splitOrder as $splitIndex => $splitValue) {
+                    switch (strtoupper($splitValue)) {
+                        case 'ORDER':
+                        case 'BY':
```

Comment:
Why not take the leading `ORDER BY` out of the parsing? It's already captured by the regex in the beginning, so we only need to parse the actual set of columns.

Also, the `ASC` and `DESC` can be only in the end of the column, so instead of splitting the column in words, can we just `preg_replace()` the trailing `ASC`|`DESC` with an empty string?

I believe the closer the parsing logic to the actual SQL syntax is, the more reliable it is in edge cases.